scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/FTPConnector.java

```diff
@@ -0,0 +1,390 @@
+        }
+
+        if (item.isFile()) {
+            // upload the file
+            String localFilePath = item.getAbsolutePath();
+            getOut().println("About to upload the file: " + localFilePath);
+            boolean uploaded = uploadSingleFile(ftpClient, localFilePath, remoteFilePath);
+            if (uploaded) {
+                getOut().println("UPLOADED a file to: " + remoteFilePath);
+            } else {
+                getOut().println("COULD NOT upload the file: " + localFilePath);
+            }
+        } else {
+            // create directory on the server
+            boolean created = ftpClient.makeDirectory(remoteFilePath);
+            if (created) {
+                getOut().println("CREATED the directory: " + remoteFilePath);
+            } else {
+                getOut().println("COULD NOT create the directory: " + remoteFilePath);
+            }
+
+            // upload the sub directory
+            String parent = Paths.get(remoteParentDir, item.getName()).toString();
+            if (remoteParentDir.equals("")) {
+                parent = item.getName();
+            }
+
+            localParentDir = item.getAbsolutePath();
+            uploadDirectory(ftpClient, remoteDirPath, localParentDir, parent);
+        }
+    }
+
+    private boolean uploadSingleFile(FTPClient ftpClient, String localFilePath, String remoteFilePath)
+            throws IOException {
+        File localFile = new File(localFilePath);
+
+        InputStream inputStream = new FileInputStream(localFile);
+        try {
+            ftpClient.setFileType(FTP.BINARY_FILE_TYPE);
+            return ftpClient.storeFile(remoteFilePath, inputStream);
+        } finally {
+            inputStream.close();
+        }
+    }
+
+    private boolean checkDirectoryExists(FTPClient ftpClient, String dirPath) throws IOException {
+        ftpClient.changeWorkingDirectory(dirPath);
+        int returnCode = ftpClient.getReplyCode();
+        if (returnCode == 550) {
+            return false;
+        }
+        return true;
+    }
+
+    private void downloadDirectory(FTPClient ftpClient, String parentDir, String currentDir, String saveDir)
+            throws IOException {
+        String dirToList = parentDir;
+        if (!currentDir.equals("")) {
+            dirToList = Paths.get(dirToList, currentDir).toString();
+        }
+        FTPFile[] subFiles = ftpClient.listFiles(dirToList);
+        if (subFiles != null && subFiles.length > 0) {
+            for (FTPFile aFile : subFiles) {
+                String currentFileName = aFile.getName();
+                if (currentFileName.equals(".") || currentFileName.equals("..")) {
+                    // skip parent directory and the directory itself
+                    continue;
+                }
+                String filePath = Paths.get(parentDir, currentDir, currentFileName).toString();
+                if (currentDir.equals("")) {
```

Comment:
I believe it is better to do currentDir.isEmpty()

Reply:
done